components/prism-core.js

```diff
@@ -280,9 +280,15 @@ var _ = _self.Prism = {
 					lookbehindLength = 0,
 					alias = pattern.alias;
 
+				if (greedy && !pattern.pattern.global) {
+					// Without the global flag, lastIndex won't work
+					pattern.pattern = RegExp(pattern.pattern.source, pattern.pattern.flags + "g");
```

Comment:
I would rather we added a separate polyfill for flags instead of refactoring our code not to use it. That way, once support is more broad, we can just ditch the polyfill.